File: src/2363.merge-similar-items.2447/solution.ts
function mergeSimilarItems (items1: number[][], items2: number[][]): number[][] {
    const map = new Map<number, number>();
    for (const [value, weight, ] of items1) {
        map.set(value, weight);
    }
    for (const [value, weight, ] of items2) {
        if (map.has(value)) {
            map.set(value, map.get(value) + weight);
        } else {
            map.set(value, weight);
        }
    }
    return [...map.entries(), ].sort((a, b) => a[0] - b[0]);
}

File: src/2363.merge-similar-items.2447/solution.java
class Solution {
    public List<List<Integer>> mergeSimilarItems(int[][] items1, int[][] items2) {
        Arrays.sort(items1,(a,b)->a[0]-b[0]);
        Arrays.sort(items2,(a,b)->a[0]-b[0]);
        List<List<Integer>> result = new ArrayList<>();
        int index1 = 0;
        int index2 = 0;
        while (index1< items1.length && index2< items2.length){
            if(items1[index1][0]<items2[index2][0]){
                result.add(List.of(items1[index1][0],items1[index1][1]));
                index1++;
            }else if(items1[index1][0]>items2[index2][0]){
                result.add(List.of( items2[index2][0], items2[index2][1] ));
                index2++;
            }else{
                result.add(List.of( items2[index2][0], items2[index2][1]+items1[index1][1] ));
                index1++;
                index2++;
            }
        }
        
        while (index1< items1.length){
            result.add(List.of(items1[index1][0],items1[index1][1]));
            index1++;
        }
        while (index2< items2.length){
            result.add(List.of( items2[index2][0], items2[index2][1] ));
            index2++;
        }
        

        return result;
    }
}